File: sdks/java/lib/src/main/java/models/shared/TrashWebLink.java
package .models.shared;

import java.time.OffsetDateTime;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import .utils.DateTimeSerializer;
import .utils.DateTimeDeserializer;

/**
 * TrashWebLink
 * Represents a trashed web link.
**/
public class TrashWebLink {
    @JsonInclude(Include.NON_ABSENT)
    @JsonSerialize(contentUsing = DateTimeSerializer.class)
    @JsonDeserialize(contentUsing = DateTimeDeserializer.class)
    @JsonProperty("created_at")
    public OffsetDateTime createdAt;
    public TrashWebLink withCreatedAt(OffsetDateTime createdAt) {
        this.createdAt = createdAt;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("created_by")
    public UserMini createdBy;
    public TrashWebLink withCreatedBy(UserMini createdBy) {
        this.createdBy = createdBy;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("description")
    public String description;
    public TrashWebLink withDescription(String description) {
        this.description = description;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("etag")
    public String etag;
    public TrashWebLink withEtag(String etag) {
        this.etag = etag;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("id")
    public String id;
    public TrashWebLink withId(String id) {
        this.id = id;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("item_status")
    public TrashWebLinkItemStatusEnum itemStatus;
    public TrashWebLink withItemStatus(TrashWebLinkItemStatusEnum itemStatus) {
        this.itemStatus = itemStatus;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonSerialize(contentUsing = DateTimeSerializer.class)
    @JsonDeserialize(contentUsing = DateTimeDeserializer.class)
    @JsonProperty("modified_at")
    public OffsetDateTime modifiedAt;
    public TrashWebLink withModifiedAt(OffsetDateTime modifiedAt) {
        this.modifiedAt = modifiedAt;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("modified_by")
    public UserMini modifiedBy;
    public TrashWebLink withModifiedBy(UserMini modifiedBy) {
        this.modifiedBy = modifiedBy;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("name")
    public String name;
    public TrashWebLink withName(String name) {
        this.name = name;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("owned_by")
    public UserMini ownedBy;
    public TrashWebLink withOwnedBy(UserMini ownedBy) {
        this.ownedBy = ownedBy;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("parent")
    public FolderMini parent;
    public TrashWebLink withParent(FolderMini parent) {
        this.parent = parent;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("path_collection")
    public TrashWebLinkPathCollectionTrash pathCollection;
    public TrashWebLink withPathCollection(TrashWebLinkPathCollectionTrash pathCollection) {
        this.pathCollection = pathCollection;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonSerialize(contentUsing = DateTimeSerializer.class)
    @JsonDeserialize(contentUsing = DateTimeDeserializer.class)
    @JsonProperty("purged_at")
    public OffsetDateTime purgedAt;
    public TrashWebLink withPurgedAt(OffsetDateTime purgedAt) {
        this.purgedAt = purgedAt;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("sequence_id")
    public java.util.Map<String, Object> sequenceId;
    public TrashWebLink withSequenceId(java.util.Map<String, Object> sequenceId) {
        this.sequenceId = sequenceId;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("shared_link")
    public String sharedLink;
    public TrashWebLink withSharedLink(String sharedLink) {
        this.sharedLink = sharedLink;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonSerialize(contentUsing = DateTimeSerializer.class)
    @JsonDeserialize(contentUsing = DateTimeDeserializer.class)
    @JsonProperty("trashed_at")
    public OffsetDateTime trashedAt;
    public TrashWebLink withTrashedAt(OffsetDateTime trashedAt) {
        this.trashedAt = trashedAt;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("type")
    public TrashWebLinkTypeEnum type;
    public TrashWebLink withType(TrashWebLinkTypeEnum type) {
        this.type = type;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("url")
    public String url;
    public TrashWebLink withUrl(String url) {
        this.url = url;
        return this;
    }
}

File: sdks/go/pkg/models/shared/trashweblink.go
package shared

import (
	"time"
)

type TrashWebLinkItemStatusEnum string

const (
	TrashWebLinkItemStatusEnumActive  TrashWebLinkItemStatusEnum = "active"
	TrashWebLinkItemStatusEnumTrashed TrashWebLinkItemStatusEnum = "trashed"
	TrashWebLinkItemStatusEnumDeleted TrashWebLinkItemStatusEnum = "deleted"
)

type TrashWebLinkPathCollectionTrashEntriesTypeEnum string

const (
	TrashWebLinkPathCollectionTrashEntriesTypeEnumFolder TrashWebLinkPathCollectionTrashEntriesTypeEnum = "folder"
)

// TrashWebLinkPathCollectionTrashEntries
// The parent folder for this item
type TrashWebLinkPathCollectionTrashEntries struct {
	Etag       *string                                         `json:"etag,omitempty"`
	ID         *string                                         `json:"id,omitempty"`
	Name       *string                                         `json:"name,omitempty"`
	SequenceID *string                                         `json:"sequence_id,omitempty"`
	Type       *TrashWebLinkPathCollectionTrashEntriesTypeEnum `json:"type,omitempty"`
}

// TrashWebLinkPathCollectionTrash
// The tree of folders that this web link is contained in,
// starting at the root.
type TrashWebLinkPathCollectionTrash struct {
	Entries    []TrashWebLinkPathCollectionTrashEntries `json:"entries"`
	TotalCount int64                                    `json:"total_count"`
}

type TrashWebLinkTypeEnum string

const (
	TrashWebLinkTypeEnumWebLink TrashWebLinkTypeEnum = "web_link"
)

// TrashWebLink
// Represents a trashed web link.
type TrashWebLink struct {
	CreatedAt      *time.Time                       `json:"created_at,omitempty"`
	CreatedBy      *UserMini                        `json:"created_by,omitempty"`
	Description    *string                          `json:"description,omitempty"`
	Etag           *string                          `json:"etag,omitempty"`
	ID             *string                          `json:"id,omitempty"`
	ItemStatus     *TrashWebLinkItemStatusEnum      `json:"item_status,omitempty"`
	ModifiedAt     *time.Time                       `json:"modified_at,omitempty"`
	ModifiedBy     *UserMini                        `json:"modified_by,omitempty"`
	Name           *string                          `json:"name,omitempty"`
	OwnedBy        *UserMini                        `json:"owned_by,omitempty"`
	Parent         *FolderMini                      `json:"parent,omitempty"`
	PathCollection *TrashWebLinkPathCollectionTrash `json:"path_collection,omitempty"`
	PurgedAt       *time.Time                       `json:"purged_at,omitempty"`
	SequenceID     map[string]interface{}           `json:"sequence_id,omitempty"`
	SharedLink     *string                          `json:"shared_link,omitempty"`
	TrashedAt      *time.Time                       `json:"trashed_at,omitempty"`
	Type           *TrashWebLinkTypeEnum            `json:"type,omitempty"`
	URL            *string                          `json:"url,omitempty"`
}
